runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/harness/FanOutStreamingEngineWorkerHarnessTest.java

```diff
@@ -242,7 +242,7 @@ public void testStreamsStartCorrectly() throws InterruptedException {
   @Test
   public void testOnNewWorkerMetadata_correctlyRemovesStaleWindmillServers()
       throws InterruptedException {
-    TestGetWorkBudgetDistributor getWorkBudgetDistributor = spy(new TestGetWorkBudgetDistributor());
+    GetWorkBudgetDistributor getWorkBudgetDistributor = mock(GetWorkBudgetDistributor.class);
```

Comment:
Without this change the test is deadlocking with this threadz https://gist.github.com/arunpandianp/c084d7389fe93202a8d9d7ba27e8ca50

Not sure what triggered it, this PR shouldn't change anything wrt locking. Any ideas?

Reply:
cc: @m-trieu

Reply:
This is a flaky test, I see it failing in other PRs too.